File: runcommand.rb
#!/usr/local/bin/ruby

require 'rubygems'
require 'net/ssh'

hosts = "hosts.txt"
commands = "commands.txt"
user = 'username'

File.open(hosts).each_line{ |host|
  host = host.chomp
  ssh = Net::SSH.start(host, user)
 File.open(commands).each_line{ |command|
    command = command.chomp
    puts "Running #{command} on #{host}.\n"
    output = ssh.exec!(command)
    puts output
 }
}

File: runcommand.pl
#!/usr/bin/perl
use warnings;
use strict;

use Net::OpenSSH;

open HOSTS,"hosts.txt";
my @hosts = <HOSTS>;
open COMMANDS,"commands.txt";
my @commands = <COMMANDS>;

for my $host (@hosts) {
  chomp ($host);
  my $ssh = Net::OpenSSH->new($host);
  $ssh->error and
    die "remote command failed: " . $ssh->error;
for my $command (@commands) {
  chomp ($command);
  print "Running $command on $host \n";
  $ssh->system($command) or
    die "remote command $command failed: " . $ssh->error;
  }
}
